sdk/dnsresolver/Azure.ResourceManager.DnsResolver/tests/Tests/DnsForwardingRulesetTests.cs

```diff
@@ -0,0 +1,156 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+using System.Threading.Tasks;
+using NUnit.Framework;
+using Azure.Core.TestFramework;
+using Azure.ResourceManager.Resources.Models;
+using Azure.Core;
+using Azure.ResourceManager.DnsResolver.Models;
+using System.Linq;
+using System.Collections.Generic;
+
+namespace Azure.ResourceManager.DnsResolver.Tests
+{
+    public class DnsForwardingRulesetTests : DnsResolverTestBase
+    {
+        private DnsResolver dnsResolver;
+        private DnsForwardingRulesetCollection dnsForwardingRulesetCollection;
+        private ResourceIdentifier outboundEndpointId;
+        private string vnetId;
+        private string subnetId;
+
+        public DnsForwardingRulesetTests(bool async) : base(async)
+        {
+        }
+
+        [SetUp]
+        public async Task CreateDnsResolverCollection()
+        {
+            var dnsResolverName = Recording.GenerateAssetName("dnsResolver-");
+            var vnetName = Recording.GenerateAssetName("dnsResolver-");
+            var dnsResolverData = new DnsResolverData(this.DefaultLocation);
+
+            vnetId = $"/subscriptions/{TestEnvironment.SubscriptionId}/resourceGroups/{TestEnvironment.ResourceGroup}/providers/Microsoft.Network/virtualNetworks/{vnetName}";
+            subnetId = $"/subscriptions/{TestEnvironment.SubscriptionId}/resourceGroups/{TestEnvironment.ResourceGroup}/providers/Microsoft.Network/virtualNetworks/{vnetName}/subnets/snet-sim2";
+
+            if (Mode == RecordedTestMode.Record)
+            {
+                await CreateVirtualNetworkAsync(vnetName);
+                await CreateSubnetAsync(vnetName);
+            }
+
+            dnsResolverData.VirtualNetwork = new WritableSubResource()
+            {
+                Id = new ResourceIdentifier(vnetId)
+            };
+
+            var subscription = await Client.GetSubscriptions().GetAsync(TestEnvironment.SubscriptionId);
+            var resourceGroup = await subscription.Value.GetResourceGroups().GetAsync(TestEnvironment.ResourceGroup);
+
+            dnsResolver = (await resourceGroup.Value.GetDnsResolvers().CreateOrUpdateAsync(true, dnsResolverName, dnsResolverData)).Value;
+            dnsForwardingRulesetCollection = resourceGroup.Value.GetDnsForwardingRulesets();
+
+            var outboundEndpointData = new OutboundEndpointData(this.DefaultLocation);
+
+            outboundEndpointData.Subnet = new WritableSubResource()
+            {
+                Id = new ResourceIdentifier(subnetId),
+            };
+
+            var outboundEndpointName = Recording.GenerateAssetName("outboundEndpoint-");
+            var outboundEndpoint = await dnsResolver.GetOutboundEndpoints().CreateOrUpdateAsync(true, outboundEndpointName, outboundEndpointData);
+
+            outboundEndpointId = outboundEndpoint.Value.Id;
+        }
+
+        [Test]
+        public async Task CreateDnsForwardingRuleset()
```

Comment:
Can we add the attribute `RecordedTest` to all the test cases?

Reply:
Update the attribute.